packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -47,7 +44,8 @@ export type Props = {
 export type Container = Element | Document;
 export type Instance = Element;
 export type TextInstance = Text;
-export type HydratableInstance = Element | Text;
+export type HydratableInstance = Instance | TextInstance;
+export type HostInstance = HydratableInstance | Container | Node;
```

Comment:
The new type `HostInstance` means all possible host node types (like comment nodes and other non-hydratable nodes) that the hydration process can potentially meet during its DOM traversal, not just `Element` (`Instance`) and `Text` (`TextInstance`).

------

❓I still haven't figured out how to make `yarn flow custom` typecheck succeed. The assumption is that `HostInstance` is at least a union of `HydratableInstance | Container`, but there is no strict guarantee that it is defined like that. Please advice, this is the only remaining thing that fails in CI now. ❓

Reply:
> The new type `HostInstance` means all possible host node types (like comment nodes and other non-hydratable nodes) that the hydration process can potentially meet during its DOM traversal, not just `Element` (`Instance`) and `Text` (`TextInstance`).
> 
> ------
> 
> ❓I still haven't figured out how to make `yarn flow custom` typecheck succeed. The assumption is that `HostInstance` is at least a union of `HydratableInstance | Container`, but there is no strict guarantee that it is defined like that. Please advice, this is the only remaining thing that fails in CI now. ❓

I decided to simplify the code and remove printing of non-hydratable nodes. This refactor resulted in removing `HostInstance` type as unnecessary, and removing `HydratableInstance` type (inlining the alias). See https://github.com/facebook/react/pull/13602/commits/3ad9c5ff4fcc385ba62689bab9d0c423b944185f description for details.